spec/index.html

```diff
@@ -857,30 +869,227 @@ <h3>Simple N3 interpretations</h3>
+    </section>
+		
+    <section>
+		<h6>String</h6>
+		<p>A literal will be considered a "string" when it has an `xs:string` datatype, a `rdf:langString` datatype due to the presence of a language tag, or when it lacks a datatype.</p>
+		
+		<p><b>Casting from string</b>: if an input value has an `xs:string` datatype that does not match the domain, it may be possible to cast the string to the domain datatype, as <a href="https://www.w3.org/TR/xpath-functions/#casting-from-strings">defined in XPath</a>. The resulting value representation must be a valid lexical form for the domain datatype.</p>
+		<p><b>Casting to string</b>: if an input value is an IRI, or any kind of literal (incl. type `xs:anyUri` or its derivations), and the domain is `xs:string`, then the value will be cast to a string as <a href="https://www.w3.org/TR/xpath-functions/#casting-to-string">defined in XPath</a> along with additional rules <a href="https://www.w3.org/TR/sparql11-query/#FunctionMapping">defined for SPARQL 1.1</a>.</p>
+    </section>
+		
+    <section>
+		<h6>Other datatypes</h6>
+		
+		<p>Other types of datatype casting may take place as <a href="https://www.w3.org/TR/xpath-functions/#casting">defined in XPath</a>. 
+
+		<p class="ednote">There is a useful chart for casting primitive types to primitive types in <a href="https://www.w3.org/TR/xpath-functions/#casting-from-primitive-to-primitive">XPath</a>, a subset of which is defined for <a href="https://www.w3.org/TR/sparql11-query/#FunctionMapping">SPARQL</a>.</p>
+    </section>
+	</section>
+	
+	<section id="builtin_operations">
+		<h4>built-in operations</h4>
+						
+		<p>Generally, depending on the built-in <a>inputs</a> and <a>outputs</a> in a <a>built-in statement</a>, the following will take place:
+			<ol>
+				<li><dfn>checking</dfn> the truthfulness of the statement, or </li>
+				<li><dfn>calculating</dfn> one or more values for the outputs so as to ensure the truthfulness</li>
+			</ol>
+		</p>
+		<p>If a built-in statement only has inputs, the truthfulness of the statement will be checked. For instance, in the statement `(2 1) math:sum 3`, the built-in `math:sum` will check whether `2 + 1 = 3`.</p>
+		<p>If a built-in statement has both inputs and outputs, one or more values can be calculated for the outputs so as to ensure its truthfulness. For instance, given `(2 1) math:sum ?s`, the N3 system will calculate value `3` for output `?s`.</p>
+		<p>These operations may include casting or substituting values to different domains, as discussed in <a href="#builtin_arg_domains">Argument domains</a>.</p>
+			
+    <section>
+		<h5>built-in theory box</h5>
+		<p>Before elaborating on these operations, we point out that a built-in statement can be more conventiently seen as a <i>constrained</i> basic graph pattern (BPG) search on the N3 built-in’s <dfn>theory box</dfn>, defined to include all truthful built-in statements for the N3 built-in.</p>
```

Comment:
```suggestion
		<p>Before elaborating on these operations, we point out that a built-in statement can, in many cases, be more conveniently seen as a <i>constrained</i> basic graph pattern (BPG) search on the N3 built-in’s <dfn>theory box</dfn>, defined to include all truthful built-in statements for the N3 built-in.</p>
```
* fix typo on "conveniently"
* add "in many cases" because SNAF built-ins such as `log:forAllIn` do not fit in the "theory box" picture, IMO
```suggestion
		<p>Before elaborating on these operations, we point out that a built-in statement can be more conventiently seen as a <i>constrained</i> basic graph pattern (BPG) search on the N3 built-in’s <dfn>theory box</dfn>, defined to include all truthful built-in statements for the N3 built-in.</p>
```

Reply:
Built-ins namespaces can be moved to Subsection 1.1 (namespaces).